### PR TITLE
➖ Remove the need for external safe compare

### LINKED ISSUE #2: safe-compare shouldn't be used in koa-shopify-auth
Node has a built-in method for it: https://nodejs.org/api/crypto.html#crypto_crypto_timingsafeequal_a_b

### PR DESCRIPTION
**Pull request by amorriscode**
## Description

Fixes #2 

The `safe-compare` was an unneeded external dependency. This PR removes it and adds a file to do it without new deps.

To test locally I built a new Shopify App and [started a local dev server](https://github.com/Shopify/quilt/tree/master/packages/koa-shopify-auth#example-app). I was able to install the app on my test store.

## Type of change

- [ ]  Patch: Bug/ Documentation fix (non-breaking change which fixes an issue or adds documentation)
- [x]  Minor: New feature (non-breaking change which adds functionality)
- [ ]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above

### CODE REVIEW

File: src/auth/safe-compare.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export default function safeCompare(stringA: string, stringB: string) {
+    const aLen = Buffer.byteLength(stringA);
+    const bLen = Buffer.byteLength(stringB);
+
+    // Turn strings into buffers with equal length
+    // to avoid leaking the length
+    const buffA = Buffer.alloc(aLen, 0, 'utf8');
+    buffA.write(stringA);
+    const buffB = Buffer.alloc(aLen, 0, 'utf8');
```
[the remark this answers is not in the document]
Comment (amorriscode) [reply src/auth/safe-compare.ts:11]: Whoops!

**Comment by amorriscode** [comment]
Thanks for catching my goofs @mkevinosullivan! 🙏

**Review by tanema — APPROVED** [review]

**Review by mkevinosullivan — APPROVED** [review]

**Comment by mkevinosullivan** [comment]
👋  Hi Anthony, I was just about to merge in the changes and create a release until I noticed there are no unit tests for the `safe-compare.ts` file.

When you get a chance, can you create `src/auth/test/safe-compare.test.ts` and add a few unit tests please?

**Comment by amorriscode** [comment]
@mkevinosullivan I added a test but had a question. Although TypeScript is being used, `safeCompare` doesn't have protection again non-string input during runtime. I didn't see any guards in the code against things like that so just thought I'd ask if you want that handled?

**Comment by mkevinosullivan** [comment]
> @mkevinosullivan I added a test but had a question. Although TypeScript is being used, `safeCompare` doesn't have protection again non-string input during runtime. I didn't see any guards in the code against things like that so just thought I'd ask if you want that handled?

It's generally a good idea, but I think it's OK given the limited usage here.

Thank you for the contribution!

**Pull request merged**